plaidml/bridge/openvino/ops/deformable_convolution.cpp

```diff
@@ -0,0 +1,605 @@
+                        .input_layout(plaidml::op::TensorLayout::NCX)
+                        .filter_layout(plaidml::op::TensorLayout::KCX)
+                        .autogroup_mode(plaidml::op::AutoGroupMode::EXPLICIT)
+                        .group_layout(plaidml::op::GroupLayout::IN_C)
+                        .group_layout(plaidml::op::GroupLayout::IN_K)
+                        .groups(static_cast<int>(G));
+      return edsl::make_tuple(result);
+    }
+    // Compute 2D DeformableConvolution.
+    if (rank == 4) {
+      // Define dim of each tensor.
+      auto H = I_shape[2], W = I_shape[3], OFF_H = OFF_shape[2], OFF_W = OFF_shape[3], F_H = F_shape[2],
+           F_W = F_shape[3];
+      // Throw exception
+      if (CI % G != 0 || CO % G != 0 || CI % DG != 0 || OFF_shape[1] % DG != 0) {
+        THROW_IE_EXCEPTION << "Incorrected shape for DeformableConvolution.";
+      }
+      // split offset along dg axis.
+      std::vector<edsl::Tensor> OFF_split_dgs;
+      for (auto i = 0; i < DG; ++i) {
+        edsl::Tensor OFF_split_dg = op::slice(OFF)
+                                        .add_dim(0, N)
+                                        .add_dim(i * OFF_C / DG, (i + 1) * OFF_C / DG)
+                                        .add_dim(0, OFF_H)
+                                        .add_dim(0, OFF_W);
+        OFF_split_dgs.push_back(OFF_split_dg);
+      }
+      // Get offsets of height and width.
+      std::vector<edsl::Tensor> offset_h_slices, offset_w_slices;
+      for (auto i = 0; i < DG; ++i) {
+        edsl::Tensor OFF_split_dg_h =
+            op::slice(OFF_split_dgs[i]).add_dim(0, N).add_dim(0, OFF_C / DG - 1, 2).add_dim(0, OFF_H).add_dim(0, OFF_W);
+        edsl::Tensor OFF_split_dg_w =
+            op::slice(OFF_split_dgs[i]).add_dim(0, N).add_dim(1, OFF_C / DG, 2).add_dim(0, OFF_H).add_dim(0, OFF_W);
+        edsl::Tensor offset_h_concat_axis_2(0), offset_w_concat_axis_2(0);
+        for (auto off_h = 0; off_h < OFF_H; ++off_h) {
+          edsl::Tensor offset_h_concat_axis_3(0), offset_w_concat_axis_3(0);
+          for (auto off_w = 0; off_w < OFF_W; ++off_w) {
+            edsl::Tensor single_offset_h = op::slice(OFF_split_dg_h)
+                                               .add_dim(0, N)
+                                               .add_dim(0, (OFF_C / DG) / 2)
+                                               .add_dim(off_h, off_h + 1)
+                                               .add_dim(off_w, off_w + 1);
+            edsl::Tensor single_offset_w = op::slice(OFF_split_dg_w)
+                                               .add_dim(0, N)
+                                               .add_dim(0, (OFF_C / DG) / 2)
+                                               .add_dim(off_h, off_h + 1)
+                                               .add_dim(off_w, off_w + 1);
+            single_offset_h = op::reshape(single_offset_h, make_tuple<int64_t>({N, 1, F_H, F_W}));
```

Comment:
Does here the reshaped size `Nx1xF_HxF_W` always equal  `Nx(OFF_C / DG) / 2` ?

Reply:
yes, In this op, OFF_C = 2 * DG * F_H * F_W;